astropy/convolution/convolve.py

```diff
@@ -442,12 +452,27 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0, crop=True,
                       "equivalent to the convolve boundary='fill'.  There is "
                       "no FFT equivalent to convolve's "
                       "zero-if-kernel-leaves-boundary", AstropyUserWarning)
-        psf_pad = True
+        if psf_pad is None:
+            psf_pad = True
+        if fft_pad is None:
+            fft_pad = True
     elif boundary == 'fill':
         # create a boundary region at least as large as the kernel
-        psf_pad = True
+        if psf_pad is False:
+            warnings.warn("psf_pad was set to {0}, which overrides the "
+                          "boundary='fill' setting.".format(psf_pad),
+                          AstropyUserWarning)
+        else:
+            psf_pad = True
+        if fft_pad is None:
+            # default is 'True' according to the docstring
+            fft_pad = True
     elif boundary == 'wrap':
+        if psf_pad:
+            raise ValueError("With boundary='wrap', psf_pad cannot be enabled.")
         psf_pad = False
+        if fft_pad:
+            raise ValueError("With boundary='wrap', fft_pad cannot be enabled.")
```

Comment:
The argument handling updates all make sense--I think they should just be committed as a bug fix.